netbox/extras/jobs.py

```diff
@@ -39,10 +40,15 @@ def run_script(self, script, request, data, commit):
 
         try:
             try:
+                # A script can modify multiple models so need to do an atomic lock on
+                # both the default database (for non ChangeLogged models) and potentially
+                # any other database (for ChangeLogged models) - choosing Device as
+                # the model to use as it has ChangeLoggingMixin
```

Comment:
I'm not sure this is something we can reasonably tackle here; it may need to be offloaded to the script itself, as it's not possible to know what models the script might mutate, or the database routers that might be in place. IMO we should err on the safe side and leave it as-is unless we can identify a more robust solution.